control/passivity.py

```diff
@@ -45,43 +71,143 @@ def ispassive(sys):
 
-    coefficents = np.vstack(matrix_list).T
+    sys_coefficents = np.vstack(sys_matrix_list).T
 
-    constants = -np.vstack((
-        np.hstack((np.zeros_like(A),  - C.T)),
-        np.hstack((- C, -D - D.T)))
-    )
+    # LMI to ensure P is positive definite
+    P_matrix_list = __P_pos_def_constraint__(n)
+    P_coefficents = np.vstack(P_matrix_list).T
+    P_constants = np.zeros((n, n))
 
+    # cost function
     number_of_opt_vars = int(
-        (state_space_size**2-state_space_size)/2 + state_space_size)
+        (n**2-n)/2 + n)
     c = cvx.matrix(0.0, (number_of_opt_vars, 1))
+    c = cvx.matrix(np.append(np.array(c), -1.0))
+
+    Gs = [cvx.matrix(sys_coefficents)] + [cvx.matrix(P_coefficents)]
+    hs = [cvx.matrix(sys_constants)] + [cvx.matrix(P_constants)]
 
     # crunch feasibility solution
     cvx.solvers.options['show_progress'] = False
-    sol = cvx.solvers.sdp(c,
-                          Gs=[cvx.matrix(coefficents)],
-                          hs=[cvx.matrix(constants)])
+    sol = cvx.solvers.sdp(c, Gs=Gs, hs=hs)
+    return sol["x"]
+
+
+def ispassive(sys, rho=None, nu=None):
+    '''
+    Indicates if a linear time invariant (LTI) system is passive
+
+    Constructs a linear matrix inequality and a feasibility optimization
+    such that if a solution exists, the system is passive.
+
+    The sources for the algorithm are: 
+
+    McCourt, Michael J., and Panos J. Antsaklis
+        "Demonstrating passivity and dissipativity using computational methods." 
 
-    return (sol["x"] is not None)
+    Nicholas Kottenstette and Panos J. Antsaklis
+        "Relationships Between Positive Real, Passive Dissipative, & Positive Systems" 
+        equation 36.
+
+    Parameters
+    ----------
+    sys: An LTI system
+        System to be checked.
+    nu: float
+        Concrete value for input passivity index. 
+    rho: float
+        Concrete value for output passivity index. 
+
+    Returns
+    -------
+    bool, float, or None: 
+        The input system is passive, or the passivity index "opposite" the input. 
+        If the problem is unfeasiable when requesting the passivity index, returns None.
+    '''
+    if rho is not None and nu is not None:
+        return __ispassive__(sys, rho, nu) is not None
+    elif rho is None and nu is not None:
+        sol = __ispassive__(sys, nu=nu)
+        if sol is not None:
+            return sol[-1]
+        else:
+            return None
+    elif nu is None and rho is not None:
+        sol = __ispassive__(sys, rho=rho)
+        if sol is not None:
+            return sol[-1]
+        else:
+            return None
+    else:
+        sol_rho = __ispassive__(sys, nu=eps)
+        sol_nu = __ispassive__(sys, rho=eps)
+        if sol_rho is not None and sol_nu is not None:
+            return sol_rho[-1] > 0 or sol_nu[-1] > 0
+        else:
+            return False
+
+
+def getPassiveIndex(sys, index_type=None):
```

Comment:
The convention in python-control is to use intercaps for class names, but not functions.  This should be `get_passive_index` (or `get_passivity_index`?).